upup/pkg/kutil/delete_cluster_vsphere.go

```diff
@@ -17,10 +17,136 @@ limitations under the License.
 package kutil
 
 import (
-	"fmt"
+	"context"
+	"github.com/golang/glog"
+	"github.com/vmware/govmomi/find"
+	"github.com/vmware/govmomi/object"
+	"github.com/vmware/govmomi/vim25/types"
+	"k8s.io/kops/upup/pkg/fi"
+	"k8s.io/kops/upup/pkg/fi/cloudup/vsphere"
 )
 
+const (
+	typeVM = "VM"
+)
+
+type clusterDiscoveryVSphere struct {
+	cloud        fi.Cloud
+	vsphereCloud *vsphere.VSphereCloud
+	clusterName  string
+}
+
+type vsphereListFn func() ([]*ResourceTracker, error)
+
 func (c *AwsCluster) listResourcesVSphere() (map[string]*ResourceTracker, error) {
```

Comment:
Agreed, the type name ```AwsCluster``` doesn't look good in delete_cluster_vsphere. This will be changed in upstream. They have todo comment which says that.